static/js/components/CustomizeOpenAIParameters.jsx

```diff
@@ -0,0 +1,238 @@
+    (state) => state.config.openai_summary_parameters
+  );
+  const user_openai_summary_parameters = useSelector(
+    (state) => state.profile.preferences.summary.OpenAI
+  );
+
+  const default_openai_summary_parameters = {
+    ...site_openai_summary_parameters,
+    ...user_openai_summary_parameters,
+  };
+
+  const handleAIClickOpen = () => {
+    setAIOpen(true);
+  };
+
+  const handleAIClose = () => {
+    setAIOpen(false);
+  };
+
+  const handleAISubmit = ({ formData }) => {
+    const prefs = {
+      summary: {
+        OpenAI: {
+          active: user_openai_summary_parameters.active,
+          ...formData,
+        },
+      },
+    };
+    dispatch(profileActions.updateUserPreferences(prefs));
+    handleAIClose();
+  };
+
+  const formSchema = {
+    type: "object",
+    properties: {
+      model: {
+        type: "string",
+        title: "model",
+        examples: [
+          "gpt-4",
+          "gpt-3.5-turbo",
+          "text-davinci-003",
+          "davinci",
+          "gpt-4-32k",
+        ],
+      },
+      prompt: {
+        type: "string",
+        title: "prompt",
+      },
+      temperature: {
+        type: "number",
+        title: "Temperature",
+        minimum: 0.0,
+        maximum: 1.0,
+        multipleOf: 0.05,
+      },
+      max_tokens: {
+        type: "integer",
+        title: "max_tokens",
+        minimum: 10,
+        maximum: 1000,
+        multipleOf: 10,
+      },
+      top_p: {
+        type: "number",
+        title: "top_p",
+        minimum: 0.0,
+        maximum: 1.0,
+        multipleOf: 0.05,
+      },
+      frequency_penalty: {
+        type: "number",
+        title: "frequency_penalty",
+        minimum: -2.0,
+        maximum: 2.0,
+        multipleOf: 0.05,
+      },
+      presence_penalty: {
+        type: "number",
+        title: "presence_penalty",
+        minimum: -2.0,
+        maximum: 2.0,
+        multipleOf: 0.05,
+      },
+    },
+    required: [
+      "temperature",
+      "max_tokens",
+      "top_p",
+      "frequency_penalty",
+      "model",
+      "presence_penalty",
+      "prompt",
+    ],
+  };
+
+  const uiSchema = {
+    temperature: {
+      "ui:widget": "updown",
+      "ui:help":
+        "What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.",
+    },
+    top_p: {
+      "ui:widget": "updown",
+      "ui:help":
+        "An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered.",
+    },
+    prompt: {
+      "ui:widget": "textarea",
+    },
+    max_tokens: {
+      "ui:widget": "range",
+      "ui:help":
+        "The maximum number of tokens to generate. Must be between 1 and 2048.",
+    },
```

Comment:
As Michael also suggested, I would make this a little more clear. I think that some people (especially non native English speakers) wouldn't be sure if this is the tokens used to generate (how many tokens SP can send to OpenAI), or the ones received. It could be said in a simpler way for the end user, mentioning words rather than token.